docs/topic-guides/secrets/secrets.rst

```diff
@@ -0,0 +1,143 @@
+file browser (E.G. JupyterLab).
+
+.. image:: ../../_static/images/secrets_selection.png
+  :width: 85%
+  :align: center
+  :alt: Select secrets to mount in a new session
+
+Click on the ``Start Session`` button to start the session with the selected
+secrets. You can now access the secrets in the session container at the
+specified path. The secrets will be stored in files with the same name.
+
+.. note::
+
+  Secrets will be mounted with the value stored at the session start time.
+  If you change the value of a secret after starting the session, you will
+  need to restart the session to apply the changes.
+
+Security Model
+--------------
+
+.. image:: ../../_static/images/secrets_encryption_decryption.gif
+  :width: 85%
+  :align: center
+  :alt: Secrets encryption scheme
+
+Renku stores secrets in its database, doubly encrypted, ensuring that no part 
+accessible from the internet other than the session has access to unencrypted 
+secrets. All secrets are encrypted at rest.
+
+Threat models we address are:
+- One of our public-facing services being breached
+- A malicious actor getting a copy of our database (for instance from a backup)
+
+We explicitly do not guard against:
+- Someone stealing your login details or login token
+- You starting a malicious session with secrets, as we can't control the code 
+that runs within a session.
+
+The Renku ``data service`` uses symmetric Fernet encryption with a key only it 
+knows to ensure all data is encrypted at rest in its database.
+For each user, a unique ``user key`` is generated. In addition, there is a 
+dedicated ``secret storage service`` which has an RSA 2048 bit ``public key`` 
+and a ``private key``, the latter of which is only known to this service. This 
+service is not accessible to the public internet.
+
+When a user stores a secret, it is first symmetrically encrypted with the 
+``user key``, using Fernet. We then generate a random ``secret key`` that is 
+encrypted using the ``public key`` and passed to the  ``secret storage 
+service``, meaning only it can decrypt the ``secret key``. The users secret is 
+then encrypted again using this ``secret key`` and stored in the database.
+At this point, the ``data service`` can't decrypt the user's secret anymore, as 
+it does not know the ``secret key``.
+
+To decrypt a secret, ``secret storage service`` gets a request from ``notebooks 
+service`` that a user would like to start a session with some secret mounted. 
+It uses its ``private key`` to get the ``secret key`` and uses this to decrypt 
+the outer layer of encryption of the secret. It then creates a Kubernetes 
+secret with the (now singly encrypted) user secret, which gets mounted in the 
+user session.
+On session start, an init container reads the mounted secrets, and uses the 
+``user key`` to undo the inner encryption. It then creates files inside the 
+session with the decrypted secret values.
+
```

Comment:
May I nitpick a bit ?
The Kubernetes secret is in fact only mounted in the init container that will decrypt its content. The session (if we are talking about the main container) will only see the volume containing the decrypted secrets.

Reply:
I'll let @Panaetius answer this technical aspect :grin:

Reply:
I've changed the description a bit (I think this comment meant to address the previous paragraph, not this one)